android/jni/com/mapswithme/maps/SearchEngine.cpp

```diff
@@ -238,23 +238,21 @@ extern "C"
   JNIEXPORT void JNICALL
   Java_com_mapswithme_maps_search_SearchEngine_nativeShowResult(JNIEnv * env, jclass clazz, jint index)
   {
-    lock_guard<mutex> guard(g_resultsMutex);
-    Result const & result = g_results.GetResult(index);
-    g_framework->PostDrapeTask([result]()
-    {
-      g_framework->NativeFramework()->ShowSearchResult(result);
-    });
+    unique_lock<mutex> guard(g_resultsMutex);
```

Comment:
Up to you, I'm planning to get rid of this mutex at all, because all this functions are called from the UI thread (I hope).